src/plugins/typedoc.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2017-present Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain
+ * a copy of the license in the LICENSE and PATENTS files in the root of this
+ * repository.
+ */
+
+import * as Typedoc from "typedoc";
+import {
+    ITsClassEntry,
+    ITsPropertyEntry,
+    ITypedocPluginData,
+} from "../client";
+import { ICompiler, IFile, IPlugin } from "./plugin";
+
+class TypedocApp extends Typedoc.Application {
+    public fromFiles(files: string[]) {
+        const expanded = this.expandInputFiles(files);
+        const project = this.convert(expanded);
+        return project.toObject();
+    }
+
+    // this tricks typedoc into working
+    get isCLI(): boolean {
+        return true;
+    }
+}
+
+export class TypedocPlugin implements IPlugin<ITypedocPluginData> {
+    public compile(files: IFile[], compiler: ICompiler) {
+        const typedoc = new TypedocApp({logger: "none"}).fromFiles(files.map((f) => f.path));
+
+        const classes: {[key: string]: ITsClassEntry} = {};
+        this.visitKind(typedoc, "Class", (classDef: any) => {
```

Comment:
turns out typedoc does NOT type this. It uses `toObject()` methods on all its classes that are all typed `any`. a bit surprising.